Spec/Auth.swift

```diff
@@ -4114,8 +4114,11 @@ class Auth : QuickSpec {
 
             // RSA8g RSA8c
             context("when using authUrl") {
-                let options = AblyTests.clientOptions()
-                options.authUrl = URL(string: echoServerAddress)!
+                let options: ARTClientOptions = {
```

Comment:
Gosh, quite a lot of boilerplate. This construction pattern got me thinking about [Kotlin's `apply` function](https://kotlinlang.org/docs/scope-functions.html#apply) - for example as used in one of other codebases [here](https://github.com/ably/ably-asset-tracking-android/blob/67afa76861f43885d1a4dbe280156b8334dc3846/common/src/main/java/com/ably/tracking/common/Ably.kt#L275).

I do wonder if there's an opportunity to make this less cumbersome. On a quick Google I found [this suggestion](https://stackoverflow.com/questions/43830927/how-to-write-a-generic-apply-function-in-swift/43831365#43831365), for example. 🤔

Reply:
I'm in two minds about it. It's neater, but it's also something for a person new to the codebase to need to understand (although it's relatively self-explanatory). I think in the interests of not going off on a bit of a tangent, I'll keep it as it is.